### PR TITLE
plugin to check if ec2 instance have exceeded the alarm threshold for…

### PR DESCRIPTION
**Pull request by fatima99s**
… cpu utilization

### CODE REVIEW

File: plugins/aws/ec2/overutilizedEC2Instance.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'EC2 CPU Alarm Threshold Exceeded',
+    category: 'EC2',
+    domain: 'Compute',
+    description: 'Identify EC2 instances that have exceeded the alarm threshold for CPU utilization.',
```
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:8]: Change it to something like
```suggestion
    description: 'Ensure EC2 instances do not exceed the alarm threshold for CPU utilization.',
```

File: plugins/aws/ec2/overutilizedEC2Instance.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    link: 'https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-cloudwatch.html',
+    recommended_action: 'Investigate the cause of high CPU utilization and consider optimizing or scaling resources.',
+    apis: ['EC2:describeInstances', 'CloudWatch:getEc2MetricStatistics'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        var cpuThreshold = 90;
```
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:19]: This should be a user input

File: plugins/aws/ec2/overutilizedEC2Instance.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+                    if (!getMetricStatistics.data.Datapoints.length) {
+                        helpers.addResult(results, 0,
+                            'CPU metric statistics are not available', region, resource);
+                    } else {
+                        var cpuDatapoints = getMetricStatistics.data.Datapoints;
+                        var cpuUtilization = cpuDatapoints[cpuDatapoints.length - 1].Average;
+                        if (cpuUtilization > cpuThreshold) {
+                            helpers.addResult(results, 2,
+                                `CPU threshold exceeded - Current CPU utilization: ${cpuUtilization}%`, region, resource);
```
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:62]: Can we change this to something like
```suggestion
                                `EC2 instance has current CPU utilization of ${cpuUtilization}% which exceeds the CPU threshold`, region, resource);
```

File: plugins/aws/ec2/overutilizedEC2Instance.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                        } else {
+                            helpers.addResult(results, 0,
+                                `CPU threshold not exceeded - Current CPU utilization: ${cpuUtilization}%`, region, resource);
```
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:65]: something like
```suggestion
                                `EC2 instance has current CPU utilization of ${cpuUtilization}% which does not exceed the CPU threshold`, region, resource);
```

File: plugins/aws/ec2/overutilizedEC2Instance.js (hunk 1 above)
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:8]: Change this to something like
```suggestion
    description: 'Ensure EC2 instances do not exceed the alarm threshold for CPU utilization.',
```

File: plugins/aws/ec2/overutilizedEC2Instance.js (hunk 2 above)
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:19]: This should be a user input

File: plugins/aws/ec2/overutilizedEC2Instance.js (hunk 3 above)
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:62]: can we change this to something like
```suggestion
                                `EC2 instance has current CPU utilization of ${cpuUtilization}% which exceeds the CPU threshold`, region, resource);
```

File: plugins/aws/ec2/overutilizedEC2Instance.js (hunk 4 above)
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:65]: change this to something like
```suggestion
                                `EC2 instance has current CPU utilization of ${cpuUtilization}% which does not exceed the CPU threshold`, region, resource);
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/aws/ec2/overutilizedEC2Instance.js (hunk 2 above)
Comment (fatima99s) [reply plugins/aws/ec2/overutilizedEC2Instance.js:19]: do i need to write settings for this?

Comment (mehakseedat63) [reply plugins/aws/ec2/overutilizedEC2Instance.js:19]: yes

File: plugins/aws/ec2/overutilizedEC2Instance.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'EC2 CPU Alarm Threshold Exceeded',
+    category: 'EC2',
+    domain: 'Compute',
+    description: 'Ensure EC2 instances do not exceed the alarm threshold for CPU utilization.',
+    more_info: 'Excessive CPU utilization can indicate performance issues or the need for capacity optimization.',
+    link: 'https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-cloudwatch.html',
```
Comment (mehakseedat63) [inline plugins/aws/ec2/overutilizedEC2Instance.js:10]:
```suggestion
    link: 'https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/viewing_metrics_with_cloudwatch.html#ec2-cloudwatch-metrics',
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**